projects/js-packages/number-formatters/README.md

```diff
@@ -1,11 +1,144 @@
+
+In the US, users will expect to see USD prices rendered with the currency symbol `$`. However, there are many other currencies which use `$` as their currency symbol (eg: `CAD`). This package tries to prevent confusion between these symbols by using an international version of the symbol when the locale does not match the currency. So if your locale is `en-CA`, USD prices will be rendered with the symbol `US$`.
+
+However, this relies on the user having set their interface language to something other than `en-US`/`en`, and many English-speaking non-US users still have that interface language (eg: there's no English locale available in our settings for Argentinian English so such users would probably still have `en`). As a result, those users will see a price with `$` and could be misled about what currency is being displayed. Setting `geoLocation` helps prevent that from happening by showing `US$` for those users.
+
+For practical usages, see the wp-calypso monorepo.
+
+## Dev
+
+### Setup
+
+We try to keep a similar functional separation with `Intl.NumberFormat( ... ).format( ... )`:
+
+```
+src/
+├── number-format-currency/index.ts
+├── number-format.ts { numberformat, numberFormatCompact }
+```
+
+- these bring back cached instances of `Intl.NumberFormat()`
+- exported on their own, they require a `locale` to be passed as a parameter (we don't currently export these, but we can down the line)
+- the above isn't the case yet for `number-format-currency`. It returns the formatted number, so it's essentially a full `Intl.NumberFormat( ... ).format( ... )` call. It may receive a refactor to sync down the line, which will hopefully improve some of the code sharing
 
-const formattedNumber = formatNumber( 42 ); // Returns "42"
 ```
+src/
+├── create-number-formatters.ts { formatNumber, formatNumberCompact, formatCurrency, etc.}
+```
+
+- a closure to handle `locale` state, returning abstractions over the number formatters from above, called `formaNumber`, `formatNumberCompact`, etc.
+- the methods accept arguments to parameterise the underlying formatters and enable conveniences over common usages (e.g. a convenient `decimals` prop)
+
+Adding a new formatter to the mix, say "formatPercentage", would basically just repeat what we've done for "compact" numbers: a method returned from `create-number-formatters.ts` and a function in `number-format.ts` (or separate file if anything deeper than a few simple defaults).
+
+### Workflow
+
+When making changes, the package can be built individually via `npm run build` and the latest version linked via npm/yarn to a respective outside repo to test e.g. wp-calypso.
```

Comment:
We use pnpm in the Jetpack monorepo, so best to recommend that.
```suggestion
When making changes, the package can be built individually via `pnpm run build` and the latest version linked via npm/yarn to a respective outside repo to test e.g. wp-calypso.
```
Or you might go a step further and recommend `jetpack build -v js-packages/number-formatters`, although that assumes the reader has set up [our CLI tool](https://github.com/Automattic/jetpack/blob/trunk/tools/cli/README.md).